### PR TITLE
refactor: refactored internal handling of external and internal events

### PR DESCRIPTION
**Pull request by JanZachmann**
- introduced `Command` enum to handle all events generically (direct methods, desired properties, web service requests)
  - introduced `struct` for all json objects in order to benefit from `serde` `serialize`/`deserialize`
  - refactored `Feature` trait:
    - `command()` is now the only method handling external input by consuming a `Command` variant
    - removed obsolete methods: `handle_event()`, `ensure()`, `as_any`, `as_any_mut`
- fixed deadlocks caused by using `block_on` by using a mutex instead
- `NetworkStatus` feature:
  - renamed feature to `Network`
  - moved `reload_network()` from system.rs to `Network` feature
- started refactoring test:
  - `SshTunnel`, `DeviceUpdateConsent` and `FactoryReset` tests: moved from mod_test.rs to Feature
  - removed unnecessary json parsing tests in mode_test.rs, since now covered by `Command::from_desired_property()` and `Command::from_direct_method()` tests
  - refactoring remaining tests in mod_test.rs will be done in a new story

### CODE REVIEW

File: src/web_service.rs [hunk 1]
```diff
@@ -331,20 +316,23 @@ pub async fn publish(channel: PublishChannel, value: serde_json::Value) -> Resul
...
             tokio::time::Instant::now() + tokio::time::Duration::from_secs(15),
             request.send(),
         )
         .await
         {
-            Err(_) => error!("timeout"),
+            Err(_) => bail!("publish_to_endpoints: timeout while sending {msg} to {}.", endpoint.url),
             Ok(Err(e)) => warn!(
-                "publish: sending request {msg} to {} failed with {e}. Endpoint not present?",
+                "publish_to_endpoints: sending {msg} to {} (failed with {e}). Endpoint not present?",
```
Comment (empwilli) [inline src/web_service.rs:332]: How is a timeout different from other types of errors that may occure after the connection was established? Why bail in the one case and log an error in the other?

File: README.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ Result:
 
 The supported reset `mode` and the documentation in general can be found in the [meta-omnect layer](https://github.com/omnect/meta-omnect#factory-reset).
 
-The **optional** `preserve` array can be used to define system resp. user settings that must be restored after wiping device storage. Supported are keys from [here](https://github.com/omnect/meta-omnect/blob/main/recipes-omnect/omnect-device-service/omnect-device-service/factory-reset.json) and the key "applications" if there is a [custom configuration file](https://github.com/omnect/meta-omnect/tree/main?tab=main#custom-factory-reset-configuration) in `/etc/omnect/factory-reset.d/`.
+The **optional** `preserve` array can be used to define system resp. user settings that must be restored after wiping device storage. Supported are keys from [here](https://github.com/omnect/meta-omnect/blob/main/recipes-omnect/omnect-device-service/omnect-device-service/factory-reset.json) and the key "applications" if there is a [custom configuration file](https://github.com/omnect/meta-omnect?tab=readme-ov-file#custom-factory-reset-configuration) in `/etc/omnect/factory-reset.d/`.
```
Comment (mlilien) [inline README.md:149]: just for me: what makes this link better as the original one?

File: README.md [hunk 3]
```diff
@@ -719,8 +719,10 @@ The web service features is disabled by default and must be explicitly activated
 
 ### Factory reset
 
+Description [Factory reset](#factory-reset).
+
 ```
-curl -X POST --unix-socket /run/omnect-device-service/api.sock http://localhost/factory-reset/v1
+curl -X POST --unix-socket /run/omnect-device-service/api.sock http://localhost/factory-reset/v1 -d '{"mode": 1, "preserve": []}'
```
Comment (mlilien) [inline README.md:725]: i prefer `--json` instead of `-d` to make it more obvious 
(it also sets the content type, which we currently seem to ignore then.)

**Review by mlilien — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (JanZachmann) [reply README.md:149]: the old one didn't work for me

Comment (mlilien) [reply README.md:149]: still? or temporary?

Comment (JanZachmann) [reply README.md:149]: mmh now the existing link works at least from github. not sure anymore but as far as I remember it didn't work from VSC readme preview...

**Review by empwilli — APPROVED** [review]

File: src/web_service.rs [hunk 4]
```diff
@@ -74,29 +74,23 @@ impl WebService {
...
-            })
-            .lock()
-            .await;
+            Mutex::new(
+                serde_json::from_reader(std::io::BufReader::new(
+                    std::fs::File::open(publish_endpoints_path!())
+                        .expect("cannot open endpoints file"),
+                ))
+                .expect("cannot parse endpoints file"),
+            )
```
[the remark this answers is not in the document]
Comment (JanZachmann) [reply src/web_service.rs:92]: created a task to impl. and test your proposal

File: src/web_service.rs [hunk 5]
```diff
@@ -74,29 +74,23 @@ impl WebService {
...
-            .await;
+        PUBLISH_CHANNEL_MAP.get_or_init(|| Mutex::new(serde_json::Map::default()));
 
-        let _unused = PUBLISH_ENDPOINTS
-            .get_or_init(|| {
-                let Ok(true) = std::path::Path::new(&publish_endpoints_path!()).try_exists() else {
-                    info!("run: no endpoint file present");
+        PUBLISH_ENDPOINTS.get_or_init(|| {
+            let Ok(true) = std::path::Path::new(&publish_endpoints_path!()).try_exists() else {
+                info!("run: no endpoint file present");
```
[the remark this answers is not in the document]
Comment (JanZachmann) [reply src/web_service.rs:81]: created a task to impl. and test your proposal

**Pull request merged**